packages/ui-causal-graph-editor/src/shared/editor-overlay/legend/legend-data.tsx

```diff
@@ -74,8 +114,8 @@ const DEFAULT_LEGENDS: Map<EditorMode, LegendLineDefinition[]> = new Map([
 
 export function getLegendData(
     editorMode: EditorMode,
-    additionalLegend: LegendLineDefinition[]
-): LegendLineDefinition[] {
+    additionalLegend: GraphLegendDefinition[]
+): GraphLegendDefinition[] {
     const modeData = DEFAULT_LEGENDS.get(editorMode) ?? [];
 
     return [...modeData, ...(additionalLegend ?? []).filter(Boolean)];
```

Comment:
What's our solution for overriding default legends? As we discussed in the past, this is fine for extra legends but what if the user has a custom color scheme for nodes/edges depending on some property they've defined and they don't want to display the default legend, just their custom one.

One option we've discussed is removing the defaults from here and moving them to the Python side so they can be fully customised